### PR TITLE
drivers: pwm: microchip: Add MEC172x PWM support

### PR DESCRIPTION
**Pull request by jvasanth1**
Add support for MEC172x PWM and pin control changes.

### CODE REVIEW

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.dts [hunk 1]
```diff
@@ -234,3 +234,9 @@
 		      &kso13_gpio126 >;
 	pinctrl-names = "default";
 };
+
+&pwm0 {
+	status = "okay";
+	pinctrl-0 = < &pwm0_gpio053 >;
```
Comment (gmarull) [inline boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.dts:240]:
```suggestion
	pinctrl-0 = <&pwm0_gpio053>;
```

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig [hunk 2]
```diff
@@ -20,3 +20,5 @@ CONFIG_I2C=y
 CONFIG_ESPI=y
 CONFIG_SPI=y
 CONFIG_SPI_ASYNC=y
+CONFIG_PWM=y
+CONFIG_PWM_XEC=y
```
Comment (gmarull) [inline boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig:24]: PWM is not a minimal peripheral, so it should not be in the board defconfig per porting guidelines (actually same applies for some other entries in that file).

File: dts/bindings/pwm/microchip,xec-pwm.yaml [hunk 3]
```diff
@@ -13,3 +13,18 @@ properties:
 
     label:
       required: true
+
+    pinctrl-0:
+      required: false
+
+    pinctrl-names:
+      required: false
```
Comment (gmarull) [inline dts/bindings/pwm/microchip,xec-pwm.yaml:21]:
```suggestion
```

this is redundant

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.dts (hunk 1 above)
Comment (jvasanth1) [reply boards/arm/mec172xevb_assy6906/mec172xevb_assy6906.dts]: Fixed

**Review by gmarull — APPROVED** [review]
pinctrl part LGTM

**Review by albertofloyd — APPROVED** [review]

**Review by mbolivar-nordic — COMMENTED** [review]
ACK on the devicetree portions

**Pull request merged**